[GPUP][MSE] incorrect handling when multiple init segments are processed during a single appendBuffer

jyavenard opened the pull request:
#### f820f4ae85677af2baa1fc8f50d354c3bfc56fc0
<pre>
[GPUP][MSE] incorrect handling when multiple init segments are processed during a single appendBuffer
<a href="https://bugs.webkit.org/show_bug.cgi?id=254079">https://bugs.webkit.org/show_bug.cgi?id=254079</a>
rdar://106859839

Reviewed by Youenn Fablet.

It is necessary to tie the Media Segment being parsed to the right Init Segment.
To do so, we queue all samples and init segments found in the input buffer
until they have all been parsed by the SourceBufferPrivate.
We processed then the next init segment if any, initialize it and await
for the initialization to complete.
Once done, we will process all Media Segment tie to this Init Segment.
Once done, we repeat these steps with the next Init Segment in the Input
Buffer if any.

Tests:
* LayoutTests/media/media-source/media-source-multiple-concurrent-initialization-segments-expected.txt: Added.
* LayoutTests/media/media-source/media-source-multiple-concurrent-initialization-segments.html: Added.

* LayoutTests/media/media-source/media-source-stpp-crash-expected.txt:
* LayoutTests/media/media-source/media-source-stpp-crash.html: Amend test to explicitly show that error event should be fired.
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::m_logIdentifier):
(WebCore::SourceBuffer::~SourceBuffer):
(WebCore::SourceBuffer::removedFromMediaSource):
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment): Streamline handling of errors,
having it all managed by sourceBufferPrivateAppendComplete. It also prevents the MediaSource to be deleted and
the SourceBuffer be detached from the SourceBufferPrivate while the method is running should there be an error.
(WebCore::SourceBuffer::sourceBufferPrivateAppendError): Deleted.
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/platform/graphics/AudioTrackPrivate.h:
(isType):
* Source/WebCore/platform/graphics/InbandTextTrackPrivate.h:
(isType):
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::~SourceBufferPrivate):
(WebCore::SourceBufferPrivate::updateHighestPresentationTimestamp):
(WebCore::SourceBufferPrivate::setBufferedRanges):
(WebCore::SourceBufferPrivate::updateBufferedFromTrackBuffers):
(WebCore::SourceBufferPrivate::appendCompleted):
(WebCore::SourceBufferPrivate::reenqueSamples):
(WebCore::SourceBufferPrivate::clearTrackBuffers):
(WebCore::SourceBufferPrivate::fastSeekTimeForMediaTime):
(WebCore::SourceBufferPrivate::provideMediaData):
(WebCore::SourceBufferPrivate::removeCodedFrames):
(WebCore::SourceBufferPrivate::evictCodedFrames):
(WebCore::SourceBufferPrivate::setClient): Simplify detection that we&apos;ve been detached
(WebCore::SourceBufferPrivate::detach): removing redundant member variable.
(WebCore::SourceBufferPrivate::isAttached const): Rely on m_client value to check if we&apos;re still attached to SourceBuffer.
(WebCore::SourceBufferPrivate::didReceiveInitializationSegment):
(WebCore::SourceBufferPrivate::didReceiveSample):
(WebCore::SourceBufferPrivate::append):
(WebCore::SourceBufferPrivate::processPendingOperations):
(WebCore::SourceBufferPrivate::abortPendingOperations):
(WebCore::SourceBufferPrivate::processInitSegment):
(WebCore::SourceBufferPrivate::processMediaSample):
(WebCore::SourceBufferPrivate::processMediaSamples):
(WebCore::SourceBufferPrivate::resetParserState):
(WebCore::SourceBufferPrivate::didReceiveSampleForTrackId): Deleted.
(WebCore::SourceBufferPrivate::processPendingSamples): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
(WebCore::SourceBufferPrivate::isMediaSampleAllowed const):
(WebCore::SourceBufferPrivate::setIsAttached): Deleted.
(WebCore::SourceBufferPrivate::processingInitializationSegment const): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivateClient.h:
(WebCore::SourceBufferPrivateClient::isAsync const):
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.h:
* Source/WebCore/platform/graphics/TrackPrivateBase.h: Allow dynamic casting for Tracks objects.
* Source/WebCore/platform/graphics/VideoTrackPrivate.h:
(isType):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::setTrackChangeCallbacks):
(WebCore::SourceBufferPrivateAVFObjC::didParseInitializationData):
(WebCore::SourceBufferPrivateAVFObjC::didProvideMediaDataForTrackId):
(WebCore::SourceBufferPrivateAVFObjC::isMediaSampleAllowed const):
(WebCore::SourceBufferPrivateAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferPrivateAVFObjC::enqueueSample):
(WebCore::SourceBufferPrivateAVFObjC::processPendingTrackChangeTasks): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::didReceiveSampleForTrackId): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::didReceiveInitializationSegment):
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h:
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.cpp:
(WebCore::MockSourceBufferPrivate::didReceiveInitializationSegment):
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp:
(WebKit::RemoteSourceBufferProxy::RemoteSourceBufferProxy):
(WebKit::RemoteSourceBufferProxy::~RemoteSourceBufferProxy):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateAppendError): Deleted.
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateAppendError): Deleted.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.messages.in: Remove unnecessary method.

Canonical link: <a href="https://commits.webkit.org/262300@main">https://commits.webkit.org/262300@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3bb70664b3b40fe9fb894ff55244b3eb5f626f94

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1085 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1115 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1154 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1708 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/960 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1070 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1165 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1195 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1157 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1093 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1031 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1011 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1587 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1056 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1008 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/996 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/991 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1038 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2086 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1035 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/969 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/977 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1018 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/290 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1053 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
```diff
@@ -542,48 +558,31 @@ void SourceBufferPrivate::updateTrackIds(Vector<std::pair<AtomString, AtomString
 void SourceBufferPrivate::setClient(SourceBufferPrivateClient* client)
 {
     ASSERT(isMainThread());
+    ASSERT(client);
```
Can we change to setClient(SourceBufferPrivateClient&)?

youennf commented on Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
```diff
@@ -614,27 +613,111 @@ bool SourceBufferPrivate::validateInitializationSegment(const SourceBufferPrivat
-        auto trackId = parseInteger<uint64_t>(originalSample->trackID());
-        ASSERT(trackId);
-        m_pendingMediaSamples.append(std::make_pair(*trackId, WTFMove(originalSample)));
+    auto& initOperation = *m_pendingAppendOperations.first().initOperation;
```
Should processInitSegment take the operation as a parameter?

jyavenard replied:
I could, and move the check logic from processInitSegment to processPendingOperation instead. However, it would need to change the logic in the completionHandler where it's now up to processPendingOperation to remove the initOperation.

I'm not sure if that would be a positive gain there.

youennf commented on Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
```diff
@@ -614,27 +613,111 @@ bool SourceBufferPrivate::validateInitializationSegment(const SourceBufferPrivat
+        processMediaSamples();
+        if (m_didReceiveSampleErrored)
+            break;
+        m_pendingAppendOperations.removeFirst();
```
Since we are removing the operation here, would it clarify things if we were taking the operation as the first step, and maybe pass it to processInitSegment or processMediaSamples?

jyavenard replied:
The initOperation isn't to be dropped until the init CompletionHandler as completed, Right now, the existence of the initOperation indicates that it hasn't completed yet. We would need to add a boolean now. 
I'll see what I can do, but I'm not sure it will end up being any better.

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
 {
-    ALWAYS_LOG(LOGIDENTIFIER);
+    for (auto& track : tracks) {
+        if (auto videoTrack = dynamicDowncast<VideoTrackPrivate>(track)) {
```
We could use switch(track.type()). Do we need to handle text variant?

jyavenard replied:
text tracks can exists, but they can't have any samples though.

jyavenard replied:
I will still need to call downcast<> to call the methods, those aren't virtual so it doesn't help. rather than using an array of the base class, I could use an array of variant, but I think it would make it all more complicated than it need to be.

youennf commented on Source/WebCore/platform/graphics/InbandTextTrackPrivate.h:
```diff
@@ -100,6 +100,7 @@ class InbandTextTrackPrivate : public TrackPrivateBase {
         : m_format(format)
     {
     }
+    Type type() const override { return Type::Text; };
```
final?

youennf commented on Source/WebCore/platform/graphics/AudioTrackPrivate.h:
```diff
@@ -89,6 +89,7 @@ class AudioTrackPrivate : public TrackPrivateBase {
 
 protected:
     AudioTrackPrivate() = default;
+    Type type() const override { return Type::Audio; }
```
final?
why protected as well?

youennf commented on Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
```diff
@@ -1083,29 +1166,33 @@ void SourceBufferPrivate::didReceiveSample(Ref<MediaSample>&& originalSample)
 }
 
-void SourceBufferPrivate::append(Ref<SharedBuffer>&& buffer)
+void SourceBufferPrivate::processMediaSamples()
```
Passing an operation as parameter would remove the ASSERT and would probably be more readable.

youennf commented on Source/WebCore/platform/graphics/VideoTrackPrivate.h:
```diff
@@ -81,6 +81,7 @@ class VideoTrackPrivate : public TrackPrivateBase {
 
 protected:
     VideoTrackPrivate() = default;
+    Type type() const override { return Type::Video; }
```
final?
why protected also?

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
+                auto videoTrackSelectedChanged = [weakThis, this, trackRef = Ref { track }, selected] {
+                    if (!weakThis)
+                        return;
+                    trackDidChangeSelected(trackRef, selected);
```
Given the lambda is one line, in the initialised case, I would tend to just call this method directly.

youennf replied:
Meaning something like:
```
if (initialized)
    trackDidChangeSelected(track, selected);
else {
    m_pendingTrackChangeTasks.append([weakThis, trackRef = Ref { track }, selected] {
       if (weakThis)
           weakThis->trackDidChangeSelected(trackRef, selected);
    });
}

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
+                    trackDidChangeEnabled(trackRef, enabled);
+                };
+                if (initialized)
+                    audioTrackEnabledChanged();
```
Ditto

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
 }
 
-void SourceBufferPrivateAVFObjC::didParseInitializationData(InitializationSegment&& segment)
+void SourceBufferPrivateAVFObjC::setTrackChangeCallbacks(const Vector<RefPtr<TrackPrivateBase>>& tracks, bool initialized)
```
Can we take a Vector<Ref<>>?

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
-                trackDidChangeEnabled(trackRef, enabled);
-            };
+        m_protectedTrackInitDataMap.swap(m_pendingProtectedTrackInitDataMap);
+        m_pendingProtectedTrackInitDataMap.clear();
```
m_protectedTrackInitDataMap = std::exchange(m_pendingProtectedTrackInitDataMap, { });?

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
-            m_pendingTrackChangeTasks.append(WTFMove(videoTrackSelectedChanged));
-        });
+    auto initCheckTask = [weakThis = WeakPtr { *this }, this, tracks, abortCalled = m_abortCalled] (InitializationSegment& segment) {
+        if (!weakThis)
```
This lambda is doing quite some stuff, I wonder whether we should not do:
```
RefPtr protectedThis = weakThis.get();
if (!protectedThis)
    return;
```

Applicable elsewhere

jyavenard replied:
at this stage, it's all in the main thread. It can't change after it's been tested.

youennf replied:
'this' can be destroyed synchronously in lambdas. Hence the safer protectedThis pattern here

youennf commented on Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.h:
```diff
@@ -80,6 +79,7 @@ class RemoteSourceBufferProxy final
     void sourceBufferPrivateDidDropSample() final;
     void sourceBufferPrivateDidReceiveRenderingError(int64_t errorCode) final;
     void sourceBufferPrivateReportExtraMemoryCost(uint64_t extraMemory) final;
+    bool isAsync() const { return true; }
```
final?

jyavenard committed:
[GPUP][MSE] incorrect handling when multiple init segments are processed during a single appendBuffer
https://bugs.webkit.org/show_bug.cgi?id=254079
rdar://106859839

Reviewed by Youenn Fablet.

It is necessary to tie the Media Segment being parsed to the right Init Segment.
To do so, we queue all samples and init segments found in the input buffer
until they have all been parsed by the SourceBufferPrivate.
We processed then the next init segment if any, initialize it and await
for the initialization to complete.
Once done, we will process all Media Segment tie to this Init Segment.
Once done, we repeat these steps with the next Init Segment in the Input
Buffer if any.

Tests:
* LayoutTests/media/media-source/media-source-multiple-concurrent-initialization-segments-expected.txt: Added.
* LayoutTests/media/media-source/media-source-multiple-concurrent-initialization-segments.html: Added.

* LayoutTests/media/media-source/media-source-stpp-crash-expected.txt:
* LayoutTests/media/media-source/media-source-stpp-crash.html: Amend test to explicitly show that error event should be fired.
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::m_logIdentifier):
(WebCore::SourceBuffer::~SourceBuffer):
(WebCore::SourceBuffer::removedFromMediaSource):
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment): Streamline handling of errors,
having it all managed by sourceBufferPrivateAppendComplete. It also prevents the MediaSource to be deleted and
the SourceBuffer be detached from the SourceBufferPrivate while the method is running should there be an error.
(WebCore::SourceBuffer::sourceBufferPrivateAppendError): Deleted.
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/platform/graphics/AudioTrackPrivate.h:
(isType):
* Source/WebCore/platform/graphics/InbandTextTrackPrivate.h:
(isType):
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::~SourceBufferPrivate):
(WebCore::SourceBufferPrivate::updateHighestPresentationTimestamp):
(WebCore::SourceBufferPrivate::setBufferedRanges):
(WebCore::SourceBufferPrivate::updateBufferedFromTrackBuffers):
(WebCore::SourceBufferPrivate::appendCompleted):
(WebCore::SourceBufferPrivate::reenqueSamples):
(WebCore::SourceBufferPrivate::clearTrackBuffers):
(WebCore::SourceBufferPrivate::fastSeekTimeForMediaTime):
(WebCore::SourceBufferPrivate::provideMediaData):
(WebCore::SourceBufferPrivate::removeCodedFrames):
(WebCore::SourceBufferPrivate::evictCodedFrames):
(WebCore::SourceBufferPrivate::setClient): Simplify detection that we've been detached
(WebCore::SourceBufferPrivate::detach): removing redundant member variable.
(WebCore::SourceBufferPrivate::isAttached const): Rely on m_client value to check if we're still attached to SourceBuffer.
(WebCore::SourceBufferPrivate::didReceiveInitializationSegment):
(WebCore::SourceBufferPrivate::didReceiveSample):
(WebCore::SourceBufferPrivate::append):
(WebCore::SourceBufferPrivate::processPendingOperations):
(WebCore::SourceBufferPrivate::abortPendingOperations):
(WebCore::SourceBufferPrivate::processInitSegment):
(WebCore::SourceBufferPrivate::processMediaSample):
(WebCore::SourceBufferPrivate::processMediaSamples):
(WebCore::SourceBufferPrivate::resetParserState):
(WebCore::SourceBufferPrivate::didReceiveSampleForTrackId): Deleted.
(WebCore::SourceBufferPrivate::processPendingSamples): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
(WebCore::SourceBufferPrivate::isMediaSampleAllowed const):
(WebCore::SourceBufferPrivate::setIsAttached): Deleted.
(WebCore::SourceBufferPrivate::processingInitializationSegment const): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivateClient.h:
(WebCore::SourceBufferPrivateClient::isAsync const):
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.h:
* Source/WebCore/platform/graphics/TrackPrivateBase.h: Allow dynamic casting for Tracks objects.
* Source/WebCore/platform/graphics/VideoTrackPrivate.h:
(isType):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::setTrackChangeCallbacks):
(WebCore::SourceBufferPrivateAVFObjC::didParseInitializationData):
(WebCore::SourceBufferPrivateAVFObjC::didProvideMediaDataForTrackId):
(WebCore::SourceBufferPrivateAVFObjC::isMediaSampleAllowed const):
(WebCore::SourceBufferPrivateAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferPrivateAVFObjC::enqueueSample):
(WebCore::SourceBufferPrivateAVFObjC::processPendingTrackChangeTasks): Deleted.
(WebCore::SourceBufferPrivateAVFObjC::didReceiveSampleForTrackId): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::didReceiveInitializationSegment):
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h:
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.cpp:
(WebCore::MockSourceBufferPrivate::didReceiveInitializationSegment):
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp:
(WebKit::RemoteSourceBufferProxy::RemoteSourceBufferProxy):
(WebKit::RemoteSourceBufferProxy::~RemoteSourceBufferProxy):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateAppendError): Deleted.
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateAppendError): Deleted.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.messages.in: Remove unnecessary method.

Canonical link: https://commits.webkit.org/262300@main
